crates/oxc_semantic/src/builder.rs

```diff
@@ -67,6 +68,7 @@ pub struct SemanticBuilder<'a> {
     // to value like
     pub namespace_stack: Vec<SymbolId>,
     current_reference_flag: ReferenceFlag,
+    pub hosting_variables: FxHashMap<ScopeId, Vec<SymbolId>>,
```

Comment:
`hosting` -> `hoisting`.

Reply:
This state feels so weird. Shall we sit on this PR for at least 2 days before merging too quickly?

Reply:
Ok

Let me explain why need it. This state replaces add binding to every hoisting scope. Since we're only inserting bindings into the top-level hoisting scopes now. This will result in us not being able to check if a variable declared after the promoted variable has been declared or not

For example:

```js
function func() {
  {
    {
      var v = 0;
    }
    let v = 0;
  }
}
```

After hoisted, will be like this:

```js
function func() {
  var v;
  {
    {
      v = 0;
    }
    let v = 0;
  }
}
```
Should throw an error “Identifier `v` has already been declared” for `let v = 0;`. If without `hoisting_varaibles` we can't do that